contracts/contracts/lib/LibGateway.sol

```diff
@@ -228,23 +245,14 @@ library LibGateway {
 
     /// @notice commit topdown messages for their execution in the subnet. Adds the message to the subnet struct for future execution
     /// @param crossMessage - the cross message to be committed
-    function commitTopDownMsg(IpcEnvelope memory crossMessage) internal {
-        GatewayActorStorage storage s = LibGatewayActorStorage.appStorage();
-        SubnetID memory subnetId = crossMessage.to.subnetId.down(s.networkName);
-
-        (bool registered, Subnet storage subnet) = getSubnet(subnetId);
-
-        if (!registered) {
-            revert NotRegisteredSubnet();
-        }
-
+    function commitTopDownMsg(Subnet storage subnet, IpcEnvelope memory crossMessage) internal {
```

Comment:
feels like we just need the subnet id as the param

Reply:
Yeah, I can pass the subnetID down, but then I would need to look up the subnet again with getSubnet(subnetId), since this function actually updates the subnet state (and has been).